### PR TITLE
set status field on crd

### LINKED ISSUE #9: Settings the `status` fields on a CRD
Inverse of the existing mapping from `applyJson` and `openAPI` spec mapping. 

For any of the TF Schema fields that are `optional` put them in `status` section of CRD.

### LINKED ISSUE #50: Encrypt sensitive values in the CRD status
Terraform computed values are stored in the CRD status. These can contain sensitive details such as a storage account connection string. We can detect whether this filed is marked as sensitive in the Terraform schema and then either omit or encrypt the value before writing it into the CRD.
My preference would be to encrypt with the user provided encryption key - this allows them to use the value if they can also decrypt the key but protects it from rogue readers.
RBAC should also be used to restrict who can read the CRD as well as an additional layer of security.

### PR DESCRIPTION
**Pull request by EliiseS**
closes #51 
closes #9 
closes #50

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ func getSchemaForType(name string, item *cty.Type) *openapi_spec.Schema {
 
 		// Handle more complex types - map, set and list
 	} else if item.IsMapType() {
+		// Question: Is this not an infinite loop?
```
Comment (lawrencegripper) [inline create.go:143]: It's a recursive loop that will run until no more maps or nested types are found so not infinite, unless I've messed something up!

Comment (EliiseS) [reply create.go:143]: Aaa, `item.MapElementType()` resolves into a string or something other than a map, I think @jjcollinge

Comment (jjcollinge) [reply create.go:143]: Yeh looks ok to me

Comment (lawrencegripper) [reply create.go:143]: np, Can we remove the comment?

File: integration_test.go [hunk 2]
```diff
@@ -118,23 +120,48 @@ var _ = Describe("When creating CRDs sequentially after resources are created",
...
+				status, ok := obj.Object["status"].(map[string]interface{})
+				Expect(err).To(BeNil())
+				if !ok {
+					return fmt.Errorf("status not found in object map")
+				}
 
-			// TODO - check other `status` properties when we're mapping them
+				// If encoded, assume encrypted
+				primaryAccessKey := status["primary_access_key"].(string)
+				_, err = base64.StdEncoding.DecodeString(primaryAccessKey)
```
Comment (lawrencegripper) [inline integration_test.go:162]: Read this before the reading the comment and was like, um base64 isn't encryption ... now understand.

File: reconcile.go [hunk 3]
```diff
@@ -518,6 +527,115 @@ func (r *TerraformReconciler) applyTerraformValueToCrdStatus(value *cty.Value, c
...
+		if a == nil && e == nil {
+			return nil, nil // We found a block
+		}
+		// An error here just indicates we couldn't find the attribute or child block in the current block - we should keep looking
+	}
+
+	return nil, fmt.Errorf("Unable to find attribute or block with the key %s", key)
+}
+
+func (r *TerraformReconciler) getValueFromCtyValue(key string, value *cty.Value, block *configschema.Block) (interface{}, error) {
```
Comment (lawrencegripper) [inline reconcile.go:563]: Nice work, fun times here - can we add a todo comment that in future would be nice to wrap some tests around the conversion? Happy to take it without as we have similar code in `create.go` which needs similar tests.

File: integration_test.go (hunk 2 above)
Comment (jjcollinge) [reply integration_test.go:162]: 😂 yh we could also run a regex to check this isn't just an encoded connection string but seemed overkill for the time being

File: reconcile.go (hunk 3 above)
Comment (jjcollinge) [reply reconcile.go:564]: Done

**Pull request merged**